src/components/App/App.tsx

```diff
@@ -1,18 +1,67 @@
-import React from 'react';
-import logo from './lipstick.png';
+import React, { useEffect, useState } from 'react';
+import logo from '../../images/lipstick.png'
 import './App.css';
 import NavBar from '../NavBar/NavBar';
 import ProductContainer from '../ProductContainer/ProductContainer';
 import { Route, Routes } from 'react-router-dom';
+import { getAllProducts } from '../../apiCalls';
+import { Product } from '../../apiTypes';
 
 function App() {
+  const [allProducts, setAllProducts] = useState<Product[]>([])
+  const [filteredProducts, setFilteredProducts] = useState<Product[]>([])
+  const [loading, setLoading] = useState(false)
+  const [error, setError] = useState<Error | string | unknown>('')
+  
+
+  const updateFilteredProducts = (brand: string, type: string | null) => {
+    !type && !brand ? setFilteredProducts([]) : 
+    setFilteredProducts(allProducts.filter(product => {
+      const brandMatch = product.brand?.includes(brand.toLowerCase())
+      const typeMatch = product.product_type === type;
+      return type? brandMatch && typeMatch : brandMatch
+    }))
+  }
+
+  useEffect(() => {
+    const apiCall = async () => {
+      setLoading(true)
+      try {
+        setAllProducts(await getAllProducts())
+        setLoading(false)
+      }catch(error) {
+        setError(error)
+        setLoading(false)
+      }
+    }
+    apiCall()
+  }, [])
+
+  // useEffect(() => {
+  //   let types = allProducts.reduce((acc: string[], curr) => {
+  //     acc.push(curr.product_type);
+  //     return acc;
+  //   }, []);
+
+  //   console.log(new Set([...types]))
+    
+
+  // }, [allProducts])
+
   return (
     <main>
-      <NavBar />
-      <Routes>
-        <Route path='/' element={<ProductContainer />} />
-        <Route path='/product/:id' element={<></>} />
-      </Routes>
+      <NavBar loading={loading} products={allProducts} updateProducts={updateFilteredProducts}/>
+      {loading ? 
+        <section className='loading-container'>
+          <img className='loading' src={logo} alt='Makeup 360 spinning logo' />
+          <p className='loading-txt'>Loading...</p> 
+        </section>
+        : 
+        <Routes>
+          <Route path='/' element={<ProductContainer filteredProducts={filteredProducts}/>} />
+          <Route path='/product/:id' element={<></>} />
```

Comment:
You will want to make sure you have a * route for handling bad routes. You should be able to check that bad routes are handled properly by checking "localhost:3000/nonsense" and "localhost:3000/product/someId/nonsense". Users should see messaging like a 404 telling them that page doesn't exist and should have a way to navigate back to the homepage.